usecases/configuration_test.go

```
@@ -0,0 +1,335 @@
+	return mua.Traits
+}
+
+func (mua mockUnitAssetWithTraits) GetName() string {
+	return mua.Name
+}
+
+func (mua mockUnitAssetWithTraits) GetServices() components.Services {
+	return mua.ServicesMap
+}
+
+func (mua mockUnitAssetWithTraits) GetCervices() components.Cervices {
+	return mua.CervicesMap
+}
+
+func (mua mockUnitAssetWithTraits) GetDetails() map[string][]string {
+	return mua.Details
+}
+
+func (mua mockUnitAssetWithTraits) Serving(w http.ResponseWriter, r *http.Request, servicePath string) {
+}
+
+func createConfigHasTraits(sys *components.System) {
+	var defaultConfig templateOut
+
+	var assetTemplate components.UnitAsset
+	for _, ua := range sys.UAssets {
+		assetTemplate = *ua
+		break
+	}
+	servicesTemplate := getServicesList(assetTemplate)
+
+	confAsset := ConfigurableAsset{
+		Name:     assetTemplate.GetName(),
+		Details:  assetTemplate.GetDetails(),
+		Services: servicesTemplate,
+	}
+
+	setTest := &components.Service{
+		ID:            1,
+		Definition:    "test",
+		SubPath:       "test",
+		Details:       map[string][]string{"Forms": {"SignalA_v1a"}},
+		Description:   "A test service",
+		RegPeriod:     45,
+		RegTimestamp:  "now",
+		RegExpiration: "45",
+	}
+	ServicesMap := &components.Services{
+		setTest.SubPath: setTest,
+	}
+	mua := &mockUnitAssetWithTraits{
+		Name:        "testUnitAsset",
+		Details:     map[string][]string{"Test": {"Test"}},
+		ServicesMap: *ServicesMap,
+		CervicesMap: nil,
+		Traits:      map[string][]string{"Trait": {"testTrait"}},
+	}
+	var muaInterface components.UnitAsset = mua
+	sys.UAssets[mua.GetName()] = &muaInterface
+	// If the asset exposes traits, serialize them and store as raw JSON
+	if assetWithTraits, ok := assetTemplate.(components.HasTraits); ok {
+		if traits := assetWithTraits.GetTraits(); traits != nil {
+			traitJSON, err := json.Marshal(traits)
+			if err == nil {
+				confAsset.Traits = []json.RawMessage{traitJSON}
+			} else {
+				fmt.Println("Warning: could not marshal traits:", err)
```

Throw a fatal error (thus halting the testing) instead of printing.